tempesta_fw/t/unit/test_sched_http.c

```diff
@@ -0,0 +1,300 @@
+	}
+
+	test_req(NULL, expect_conn);
+
+	cleanup_cfg();
+	test_conn_release_all(sg);
+	test_sg_release_all();
+}
+
+TEST(tfw_sched_http, some_rules)
+{
+	TfwSrvGroup *sg1, *sg2, *sg3, *sg4, *sg5, *sg6, *sg7, *sg8, *sg9, *sg10;
+	TfwServer *srv;
+	TfwSrvConnection *expect_conn1, *expect_conn2, *expect_conn3, *expect_conn4, *expect_conn5,
+	                 *expect_conn6, *expect_conn7, *expect_conn8, *expect_conn9, *expect_conn10;
+
+	sg1 = test_create_sg("sg1", "dummy");
+	srv = test_create_srv("127.0.0.1", sg1);
+	expect_conn1 = test_create_conn((TfwPeer *)srv);
+
+	sg2 = test_create_sg("sg2", "dummy");
+	srv = test_create_srv("127.0.0.1", sg2);
+	expect_conn2 = test_create_conn((TfwPeer *)srv);
+
+	sg3 = test_create_sg("sg3", "dummy");
+	srv = test_create_srv("127.0.0.1", sg3);
+	expect_conn3 = test_create_conn((TfwPeer *)srv);
+
+	sg4 = test_create_sg("sg4", "dummy");
+	srv = test_create_srv("127.0.0.1", sg4);
+	expect_conn4 = test_create_conn((TfwPeer *)srv);
+
+	sg5 = test_create_sg("sg5", "dummy");
+	srv = test_create_srv("127.0.0.1", sg5);
+	expect_conn5 = test_create_conn((TfwPeer *)srv);
+
+	sg6 = test_create_sg("sg6", "dummy");
+	srv = test_create_srv("127.0.0.1", sg6);
+	expect_conn6 = test_create_conn((TfwPeer *)srv);
+
+	sg7 = test_create_sg("sg7", "dummy");
+	srv = test_create_srv("127.0.0.1", sg7);
+	expect_conn7 = test_create_conn((TfwPeer *)srv);
+
+	sg8 = test_create_sg("sg8", "dummy");
+	srv = test_create_srv("127.0.0.1", sg8);
+	expect_conn8 = test_create_conn((TfwPeer *)srv);
+
+	sg9 = test_create_sg("sg9", "dummy");
+	srv = test_create_srv("127.0.0.1", sg9);
+	expect_conn9 = test_create_conn((TfwPeer *)srv);
+
+	sg10 = test_create_sg("sg10", "dummy");
+	srv = test_create_srv("127.0.0.1", sg10);
+	expect_conn10 = test_create_conn((TfwPeer *)srv);
+
+	if (parse_cfg("sched_http_rules {\nmatch sg1 uri eq /foo;\n\
+	                                   match sg2 uri prefix /foo/bar;\n\
+	                                   match sg3 host eq natsys-lab.com;\n\
+	                                   match sg4 host prefix natsys-lab;\n\
+	                                   match sg5 hdr_host eq google.com;\n\
+	                                   match sg6 hdr_host prefix google;\n\
+	                                   match sg7 hdr_conn eq close;\n\
+	                                   match sg8 hdr_conn prefix Keep;\n\
+	                                   match sg9 hdr_raw eq User-Agent:Bot;\n\
+	                                   match sg10 hdr_raw prefix X-Forwarded-For;\n}\n")) {
+		TEST_FAIL("can't parse rules\n");
+	}
+
+	test_req("GET http://natsys-lab.com/foo HTTP/1.1\r\n\r\n", expect_conn1);
+	test_req("GET http://natsys-lab.com/foo/bar/ HTTP/1.1\r\n\r\n", expect_conn2);
+	test_req("GET http://natsys-lab.com/foo/baz/ HTTP/1.1\r\n\r\n", expect_conn3);
+	test_req("GET http://natsys-lab2.com/foo/baz/ HTTP/1.1\r\n\r\n", expect_conn4);
+	test_req("GET http://google.com/foo/baz/ HTTP/1.1\r\nHost: google.com\r\n", expect_conn5);
```

Comment:
> Some lines are longer than 80 character positions.
This is does not increase readability.
(https://raw.githubusercontent.com/krizhanovsky/NatSys-Lab/master/CodingStyle)